040 login system

Wojciech-Skarbek-K opened the pull request.

Wojciech-Skarbek-K committed:
added:
- user login draft

Wojciech-Skarbek-K committed:
added:
- user login draft

Wojciech-Skarbek-K committed:
Merge remote-tracking branch 'origin/main' into 040-login-system

Wojciech-Skarbek-K committed:
added:
- login endpoint test

Wojciech-Skarbek-K committed:
added:
- login endpoint test

Wojciech-Skarbek-K committed:
changed:
- backend now uses sha256 instead of bcrypt

Wojciech-Skarbek-K committed:
changed:
- backend now uses sha256 instead of bcrypt

Wojciech-Skarbek-K committed:
changed:
- validator now checks if password is hexadecimal string (sha256)

Wojciech-Skarbek-K committed:
changed:
- validator now checks if password is hexadecimal string (sha256)

Wojciech-Skarbek-K committed:
added:
- loginUser function. for now it returns 0 for incorrect login/pass and 1 for correct one.
- tests for loginUser

Wojciech-Skarbek-K committed:
added:
- loginUser function. for now it returns 0 for incorrect login/pass and 1 for correct one.
- tests for loginUser

Wojciech-Skarbek-K committed:
Merge remote-tracking branch 'origin/024-registration-system-sha256' into 040-login-system

Wojciech-Skarbek-K committed:
added:
- login endpoint now returns JWTToken on successful login

Wojciech-Skarbek-K committed:
added:
- login endpoint now returns JWTToken on successful login

Wojciech-Skarbek-K committed:
fixed:
- login endpoint now return error message correctly

Wojciech-Skarbek-K committed:
fixed:
- login endpoint now return error message correctly

Wojciech-Skarbek-K committed:
changed:
- loginUser endpoint now also returns role
- updated test for JWTToken for loginUser

Wojciech-Skarbek-K committed:
changed:
- loginUser endpoint now also returns role
- updated test for JWTToken for loginUser

shaunganley commented on src/main/java/org/kainos/groupA/GroupAConfiguration.java:
```diff
@@ -14,6 +16,10 @@ public class GroupAConfiguration extends io.dropwizard.Configuration {
 
+    //TODO: Hide it in secrets maybe?
+    @NotEmpty
+    private String jwtTokenSecret = "dfwzsdzwh823zebdwdz772632gdsbd";
```
yes please store your secret as an environment variable

shaunganley commented on src/main/java/org/kainos/groupA/dao/UserDao.java:
```diff
@@ -43,4 +64,43 @@ public int createUser(User user, Connection c) throws SQLException {
+                jws.setKey(new HmacKey(tokenSecret));
+                Map<String, String> map = new HashMap<>();
+                map.put("token", jws.getCompactSerialization());
+                map.put("role", rs.getString(2));
```
the role should be stored in the token, you don't need to return a map here, you should just return an encrypted JWT token as a string

shaunganley commented on src/main/java/org/kainos/groupA/dao/UserDao.java:
```diff
@@ -43,4 +64,43 @@ public int createUser(User user, Connection c) throws SQLException {
+
+
+    @Override
+    public Optional<LoginUser> authenticate(JwtContext jwtContext) throws AuthenticationException {
```
looks like this can be removed?

shaunganley commented on src/main/java/org/kainos/groupA/services/UserService.java:
```diff
@@ -19,4 +23,8 @@ public UserService(DatabaseConnector databaseConnector, UserDao userDao) {
         return userDao.createUser(user, databaseConnector.getConnection());
     }
+
+    public Map<String, String> loginUser(LoginUser loginUser) throws SQLException, JoseException, InvalidUserException {
```
should return JWT token as a string rather than a map

shaunganley reviewed (changes requested):
small changes required

Wojciech-Skarbek-K committed:
changed:
- merged email and name in loginuser

Wojciech-Skarbek-K committed:
changed:
- merged email and name in loginuser

Wojciech-Skarbek-K committed:
changed:
- merged email and name in loginuser

Wojciech-Skarbek-K committed:
changed:
- merged email and name in loginuser

Wojciech-Skarbek-K committed:
added:
- AuthUser (required for now for JwtAuthenticator)
- JwtAuthenticator (required by Dropwizard JWT library)

changed:
- jwtTokenSecret is now hidden in .zshrc / secrets
- loginUser now returns string instead of map

Wojciech-Skarbek-K committed:
added:
- AuthUser (required for now for JwtAuthenticator)
- JwtAuthenticator (required by Dropwizard JWT library)

changed:
- jwtTokenSecret is now hidden in .zshrc / secrets
- loginUser now returns string instead of map

Wojciech-Skarbek-K committed:
changed:
- fixed maven error about malformed project

Wojciech-Skarbek-K committed:
Merge remote-tracking branch 'origin/040-login-system' into 040-login-system

Wojciech-Skarbek-K committed:
changed:
- fixed maven error about malformed project

Wojciech-Skarbek-K committed:
changed:
- updated maven.yml to use jwttokensecret from github secrets

Wojciech-Skarbek-K committed:
Merge remote-tracking branch 'origin/main' into 040-login-system

# Conflicts:
#	src/main/java/org/kainos/groupA/GroupAApplication.java
